source/boundary_temperature/constant.cc

```diff
@@ -145,6 +145,12 @@ namespace aspect
                 {
                   boundary_id
                     = this->get_geometry_model().translate_symbolic_boundary_name_to_id (parts[0]);
+
+                  AssertThrow((this->get_fixed_temperature_boundary_indicators().find(boundary_id) != this->get_fixed_temperature_boundary_indicators().end()),
```

Comment:
I had to look at this for a little bit because you throw an exception inside a `try`-`catch` block. Would you mind moving it after the block? The `try`-`catch` block simply tries to deal with potential failure of `translate_symbolic_boundary_name_to_id`, but the error you're producing here is logically unrelated.

Reply:
Done!